btk/survey.py

```diff
@@ -1,145 +1,77 @@
-
-
-def get_survey_from_cfg(survey_conf: OmegaConf):
-    """Creates the corresponding `btk.survey.Survey` object using the information from config file.
+def get_surveys(names="LSST", psf_func: Callable = None):
+    """Return specified surveys from galcheat extended to contain PSF information.
 
     Args:
-        cfg (OmegaConf): Hydra configuration object corresponding to a single survey.
+        names (str or list): A single str specifying a survey from conf/surveys or a list with
+            multiple survey names.
+        psf_func (function): Python function which takes in two arguments: `survey` and `filter`
+            that returns a PSF as a galsim object or as a callable with no arguments.
+            If `None`, the default PSF for the specified survey will be used in each band.
 
     Returns:
-        btk.survey.Survey object.
+        galcheat.survey.Survey object or list of such objects.
     """
-    OmegaConf.resolve(survey_conf)  # in-place
-
-    survey_dict = dict(survey_conf)
-    filters_dict = dict(survey_dict.pop("filters"))
-    filter_names = list(filters_dict.keys())
-    for key in filter_names:
-        filter_dict = dict(filters_dict.pop(key))  # need to make it a dict not DictConfig
-        psf_dict = dict(filter_dict.pop("psf"))
-        if "type" not in psf_dict:
-            raise AttributeError(
-                f"Your configuration for the PSF in filter {key} in survey {survey_dict['name']}"
-                f"does not have a 'type' field which is required."
-            )
-        if psf_dict["type"] == "default":
-            psf = get_psf(**psf_dict["params"])
-        elif psf_dict["type"] == "galsim":
-            galsim_dict = dict(psf=psf_dict["params"])
-            psf = galsim.config.BuildGSObject(galsim_dict, "psf")
-        else:
-            raise NotImplementedError(
-                f"The 'type' specified for the PSF in filter {key} in "
-                f"survey {survey_dict['name']} is not implemented."
-            )
+    if isinstance(names, str):
+        names = [names]
+    if not isinstance(names, list):
+        raise TypeError("Argument 'names' of `get_surveys` should be a str or list.")
 
-        filter_dict["psf"] = psf
-        filters_dict[key] = filter_dict
-    filters = [Filter(**filters_dict[key]) for key in filters_dict]
-    survey = Survey(**survey_dict, filters=filters)
-    return survey
+    # add PSF to filters
+    surveys = []
+    for survey_name in names:
+        survey = galcheat.get_survey(survey_name)
+        for band in survey.available_filters:
+            filtr = survey.get_filter(band)
+            if psf_func is None:
+                psf = get_default_psf_with_galcheat_info(survey, filtr)
+            else:
+                psf = psf_func(survey, filtr)
+            filtr.psf = psf
+        surveys.append(survey)
 
+    if len(surveys) == 1:
+        surveys = surveys[0]
+    return surveys
```

Comment:
Very difficult to handle a function that can return both a list and a single value.
You might want to chose one return type only (maybe in a future PR)

Reply:
See #280